polling_stations/apps/data_importers/base_importers.py

```diff
@@ -0,0 +1,474 @@
+"""
+Defines the base importer classes to implement
+"""
+import abc
+import json
+import glob
+import logging
+import os
+
+from argparse import ArgumentTypeError
+from django.apps import apps
+from django.core.management.base import BaseCommand
+from django.conf import settings
+from django.contrib.gis.geos import Point, GEOSGeometry
+
+from addressbase.models import UprnToCouncil
+from councils.models import Council
+from data_importers.data_types import AddressList, StationSet
+from data_collection.data_quality_report import (
+    DataQualityReportBuilder,
+    StationReport,
+    DistrictReport,
+    ResidentialAddressReport,
+)
+from data_collection.contexthelpers import Dwellings
+from data_collection.filehelpers import FileHelperFactory
+from data_collection.loghelper import LogHelper
+from data_collection.s3wrapper import S3Wrapper
+from pollingstations.models import PollingStation
+from data_collection.models import DataQuality
+
+
+class CsvMixin:
+    csv_encoding = "utf-8"
+    csv_delimiter = ","
+
+    def get_csv_options(self):
+        return {"csv_encoding": self.csv_encoding, "csv_delimiter": self.csv_delimiter}
+
+
+class BaseImporter(BaseCommand, metaclass=abc.ABCMeta):
+
+    """
+    Turn off auto system check for all apps
+    We will maunally run system checks only for the
+    'data_collection' and 'pollingstations' apps
+    """
+
+    requires_system_checks = False
+
+    srid = 27700
+    council_id = None
+    base_folder_path = None
+    logger = None
+    batch_size = None
+    imports_districts = False
+    use_postcode_centroids = False
+
+    def write_info(self, message):
+        if self.verbosity > 0:
+            self.stdout.write(message)
+
+    def add_arguments(self, parser):
+        def check_positive(value):
+            ivalue = int(value)
+            if ivalue < 1:
+                raise ArgumentTypeError("%s is an invalid positive int value" % value)
+            return ivalue
+
+        parser.add_argument(
+            "-b",
+            "--batch_size",
+            help="<Optional> Number of records to insert in each batch when importing addresses",
+            type=check_positive,
+            required=False,
+            default=3000,
+        )
+
+        parser.add_argument(
+            "--nochecks",
+            help="<Optional> Do not perform validation checks or display context information",
+            action="store_true",
+            required=False,
+            default=False,
+        )
+
+        parser.add_argument(
+            "-p",
+            "--use-postcode-centroids",
+            help="<optional> Use postcode centroids to derive a location for polling stations",
+            action="store_true",
+            required=False,
+            default=False,
+        )
+
+    def teardown(self, council):
+        PollingStation.objects.filter(council=council).delete()
+        UprnToCouncil.objects.filter(lad=council).update(polling_station_id="")
```

Comment:
:+1: In working on this, its might also be really useful to have an updated version of `teardown.py` (although maybe slightly less-so with this data model?)

Reply:
Yes it would be handy. I've been doing a lot of `UPDATE public.addressbase_uprntocouncil
SET polling_station_id='' WHERE lad='someid'`